compiler/rustc_middle/src/mir/query.rs

```diff
@@ -149,8 +149,17 @@ pub enum ConstraintCategory<'tcx> {
     /// A constraint that doesn't correspond to anything the user sees.
     Internal,
 
-    /// An internal constraint derived from an illegal universe relation.
-    IllegalUniverse,
+    /// An internal constraint derived from an illegal placeholder relation
+    /// to this region. The arguments are a source -> drain of a path
+    /// that caused the problem, used when reporting errors.
+    IllegalPlaceholder(
```

Comment:
I don't quite like this name as it seems more 'serious' than it should be, maybe `OutlivesUnnameablePlaceholder`?

Reply:
I've had the same feeling but couldn't come up with anything. That's better, thanks!